PA193_mnemonic_Slytherin/mnemonic.py

```diff
@@ -187,6 +187,49 @@ def __init__(self, mnemonic: str):
                              .format(checksum_included, checksum_computed))
         self = mnemonic
 
+    @staticmethod
```

Comment:
why should there be this as a static method? every mnemonic necesarily computes its entropy upon instantiation, to verify validity. Moreover 'checksum' returns the checksum of Entropy, it is not checksum of the mnemonic.

Reply:
This function is not dependent on any instance of Mnemonic. It takes mnemonic phrase as string and computes checksum based on words from the mnemonic phrase and given dictionary. It is used to test `Entropy.checksum()`, because `Entropy.checksum()` computes checksum using hashing bytes of entropy. Based on our testing data, we know which mnemonic phrase corresponds to which entropy bytes, therefore we can test `Entropy.checksum()`, but we need a way how to get checksum based on mnemonic phrase **without generating entropy and hashing entropy bytes to get checksum**.

Reply:
but you are talking about tests, we can test entropy.checksum without this, we just need pairs of checksum - entropy, same as we dont have two functions to generate seed from mnemonic and still can test it. I dont think this method provides any functionality just duplicates existing one and existing code

Reply:
> we can test entropy.checksum without this, we just need pairs of checksum - entropy

But we don't have those pairs, that's the reason for this method. 1) We can extract checksum from *mnemonic phrase*. 2) We can calculate checksum from `Entropy`.

> we dont have two functions to generate seed from mnemonic and still can test it

That's different, because we already have testing data with related `Mnemonic`, `Entropy`, and `Seed`. 

> duplicates existing one and existing code

This points out, that access to dictionary hidden inside `__init__` is not a good design, because the dictionary cannot be reused and cannot be reached from different contexts than `__init__`.

How do you suggest we would test `Entropy.checksum()` without this method?

Reply:
you can use such a function to generate the test data, and then test it the same way as public methods. But I dont think it should be present in the library. Maybe it could be part of the tests, but even then it wouldnt make sense to me as, it generates the checksum the same way, just in different function.

> This points out, that access to dictionary hidden inside __init__ is not a good design, because the dictionary cannot be reused and cannot be reached from different contexts than __init__.

It might not be a good design for other reasons, but I dont think this is the case, you could duplicate any object code this way.

Reply:
> it generates the checksum the same way, just in different function

It does not, that's the point. It *"computes checksum based on words from the mnemonic phrase and given dictionary"*. On the other hand *"`Entropy.checksum()` computes checksum using hashing bytes of entropy"*. https://github.com/mvondracek/PA193_mnemonic_Slytherin/pull/41#discussion_r339294934

I can move it out from `mnemonic` to tests so it's not part of `Mnemonic` interface.

Reply:
> It does not, that's the point. It "computes checksum based on words from the mnemonic phrase and given dictionary". On the other hand "Entropy.checksum() computes checksum using hashing bytes of entropy". #41 (comment)

how is it not the same code? you literally marked it as '# region TODO copied from _DictionaryAccess.__init__' and '# region TODO copied from Mnemonic.__init__'. I mean if you called Mnemonic.init and Mnemonic.checksum with the same string as argument, it would do the same things, no? I am not saying it generates checksum same as Entropy.checksum, but it does generate it the exact same way as it is done in Mnemonic.init

Reply:
`Mnemonic.__init__` and `Mnemonic.checksum` do internally similar things. `Mnemonic.checksum` gets checksum in different way than `Entropy.checksum`. Before this method, **you could not extract checksum from `Mnemonic`**. Benefit of this method is that we could test `Entropy.checksum`. Anyway, I just moved the code to tests

Reply:
Let's me know if this is solved from your view and whether this PR could be merged or if you have any other feedback.